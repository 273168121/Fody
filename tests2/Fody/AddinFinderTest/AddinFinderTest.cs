using System.IO;
using System.Linq;
using Xunit;
#if NET46 // TODO: Remove when ObjectApproval supports .NET Core
using ObjectApproval;
#endif

public partial class AddinFinderTest : TestBase
{
    [Fact]
    public void WithNuGetPackageRoot()
    {
        var combine = Path.GetFullPath(Path.Combine(AssemblyLocation.CurrentDirectory, "Fody/FakeNuGetPackageRoot"));
        var nuGetPackageRoot = Path.GetFullPath(combine);
        var result = AddinFinder.ScanDirectoryForPackages(nuGetPackageRoot)
            .Select(s => s.Replace(@"\\", @"\").Replace(combine, ""))
            .ToList();
#if NET46 // TODO: Remove when ObjectApproval supports .NET Core
        ObjectApprover.VerifyWithJson(result);
#endif
    }

    [Fact]
    public void Integration_OldNugetStructure()
    {
        var combine = Path.GetFullPath(Path.Combine(AssemblyLocation.CurrentDirectory, "Fody/AddinFinderTest/OldNugetStructure"));
        Verify(combine);
    }

    [Fact]
    public void Integration_NewNugetStructure()
    {
        var combine = Path.GetFullPath(Path.Combine(AssemblyLocation.CurrentDirectory, "Fody/AddinFinderTest/NewNugetStructure"));
        Verify(combine);
    }

    [Fact]
    public void Integration_PaketStructure()
    {
        var combine = Path.GetFullPath(Path.Combine(AssemblyLocation.CurrentDirectory, "Fody/AddinFinderTest/PaketStructure"));
        Verify(combine);
    }

    static void Verify(string combine)
    {
        var addinFinder = new AddinFinder(
            log: s => { },
            solutionDirectory: Path.Combine(combine, "Solution"),
            msBuildTaskDirectory: Path.Combine(combine, "MsBuildDirectory/1/2/3"),
            nuGetPackageRoot: Path.Combine(combine, "NuGetPackageRoot"),
            packageDefinitions: null);
        addinFinder.FindAddinDirectories();
#if NET46 // TODO: Remove when ObjectApproval supports .NET Core
        ObjectApprover.VerifyWithJson(addinFinder.FodyFiles.Select(x => x.Replace(combine, "")));
#endif
    }
}