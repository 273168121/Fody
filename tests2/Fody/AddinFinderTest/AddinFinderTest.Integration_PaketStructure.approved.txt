[
  "\\Solution\\Packages\\Weaver7.Fody\\netclassicweaver\\Weaver7.Fody.dll",
  "\\Solution\\Packages\\Weaver8.Fody\\Weaver8.Fody.dll",
  "\\MsBuildDirectory\\Weaver1.Fody\\Weaver1.Fody.dll",
  "\\MsBuildDirectory\\Weaver2.Fody\\netclassicweaver\\Weaver2.Fody.dll",
  "\\Solution\\Tools\\Weaver.Fody.dll",
  "\\Solution\\Tools\\Weaver10.Fody\\Weaver10.Fody.dll",
  "\\Solution\\Tools\\Weaver9.Fody\\netclassicweaver\\Weaver9.Fody.dll",
  "\\NuGetPackageRoot\\Weaver5.Fody.1.0.0\\Weaver5.Fody.dll",
  "\\NuGetPackageRoot\\Weaver6.Fody.1.0.0\\netclassicweaver\\Weaver6.Fody.dll",
  "\\NuGetPackageRoot\\Weaver3.Fody\\Weaver3.Fody.dll",
  "\\NuGetPackageRoot\\Weaver4.Fody\\netclassicweaver\\Weaver4.Fody.dll"
]